tap_stripe/__init__.py

```diff
@@ -1034,24 +1067,13 @@ def sync():
     # Loop over streams in catalog
     for catalog_entry in Context.catalog['streams']:
         stream_name = catalog_entry['tap_stream_id']
-        # Sync records for stream
         if Context.is_selected(stream_name):
-            if not Context.is_sub_stream(stream_name):  # Run the sync for parent-streams
-                sync_stream(stream_name)
-                # This prevents us from retrieving 'events.events'
+            # Run the sync for only parent streams/only child streams/both parent-child streams
+            if not Context.is_sub_stream(stream_name) or not is_parent_selected(stream_name):
+                sync_stream(stream_name, Context.is_sub_stream(stream_name))
+                # This prevents us from retrieving immutable stream events.
                 if STREAM_TO_TYPE_FILTER.get(stream_name):
-                    bookmark_value = get_bookmark_for_stream(stream_name + '_events', 'updates_created')
```

Comment:
Previosuly in the older merged branch, we were fetching parent records twice when both parent and child were selected. Updated and optimized the code, hence removed this condition.